File: HW2_hash_practice.py
path='hw2_data.txt' #輸入路徑
f = open(path) #開txt

dictionary={} #字典
for line in f.readlines():#對於每一行
    try:#字典值存在則加1
        dictionary[line]+=1
    except:#字典值不存在則加入
        dictionary[line]=1

for key in dictionary:#印出結果
    print(f'{key}出現{dictionary[key]}次')
f.close #關txt

File: HW2_hash_practice.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include<map>
using namespace std;

int main() {
    ifstream ifs("hw2_data.txt", std::ios::in);//開txt

    map<string, int> dictionary;//字典

    string line;//宣告單行
    while (getline(ifs, line)) {//讀取單行
        dictionary[line]+=1;
    }
    for (const auto& word : dictionary) {//迴圈輸出字典
        std::cout << word.first << ": " << word.second << "\n";
    }
    ifs.close();
}
